### PR TITLE
fix: [CDS-70213]: Handle runtimeinput values while fetching builds for artifactory

### PR DESCRIPTION
**Pull request by vivekkarnatiharness**
## Describe your changes
image

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/50014)

### CODE REVIEW

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java [hunk 1]
```diff
@@ -230,6 +231,13 @@ private void handleErrorResponseData(DelegateResponseData responseData, String e
   public ArtifactoryResponseDTO getBuildDetails(IdentifierRef artifactoryConnectorRef, String repositoryName,
       String artifactPath, String repositoryFormat, String artifactRepositoryUrl, String orgIdentifier,
       String projectIdentifier) {
+    if (RUN_TIME_INPUT_EXPRESSION.equals(repositoryName)) {
```
Comment (sribalij) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java:234]: Since you are checking `.equals` won't it fail if it has something like "<+input>.regex("")"?

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java [hunk 2]
```diff
@@ -230,6 +231,13 @@ private void handleErrorResponseData(DelegateResponseData responseData, String e
...
+      throw new InvalidRequestException("Repository is required to fetch build details");
+    }
+
+    if (RUN_TIME_INPUT_EXPRESSION.equals(artifactPath)) {
```
Comment (sribalij) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java:238]: Same as above

**Review by sribalij — COMMENTED** [review]
Added comments about regex condition

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java (hunk 1 above)
Comment (vivekkarnatiharness) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java]: Done,fixed it

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java (hunk 2 above)
Comment (vivekkarnatiharness) [reply 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java]: Done fixed it!

**Comment by vivekkarnatiharness** [comment]
trigger uts

File: 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java [hunk 3]
```diff
@@ -230,6 +231,13 @@ private void handleErrorResponseData(DelegateResponseData responseData, String e
   public ArtifactoryResponseDTO getBuildDetails(IdentifierRef artifactoryConnectorRef, String repositoryName,
       String artifactPath, String repositoryFormat, String artifactRepositoryUrl, String orgIdentifier,
       String projectIdentifier) {
+    if (repositoryName.contains(RUN_TIME_INPUT_EXPRESSION)) {
+      throw new InvalidRequestException("Repository is required to fetch build details");
+    }
```
Comment (Abhishek-CDC) [inline 125-cd-nextgen/modules/common/core/src/main/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImpl.java:236]: Can repositoryName be null or empty?
We have methods like 'validateIfAnyValueAssigned', 'validateIfAllValuesAssigned' in 'ArtifactUtils' class, checking for null, empty and runtimePattern values.
Please refer method 'getSuccessfulBuild' in 'DockerResourceServiceImpl' class

**Review by puthrayaharness — DISMISSED** [review]

File: 125-cd-nextgen/src/test/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImplTest.java [hunk 4]
```diff
@@ -330,6 +330,23 @@ public void testGetBuildDetails2() {
...
+                                      .projectIdentifier(PROJECT_IDENTIFIER)
+                                      .orgIdentifier(ORG_IDENTIFIER)
+                                      .build();
+
+    assertThatThrownBy(()
+                           -> artifactoryResourceService.getBuildDetails(identifierRef, "<+input>", IMAGE_PATH,
+                               RepositoryFormat.docker.name(), null, ORG_IDENTIFIER, PROJECT_IDENTIFIER))
+        .isInstanceOf(InvalidRequestException.class)
+        .hasMessage("value for repository is empty or not provided");
+  }
```
Comment (Abhishek-CDC) [inline 125-cd-nextgen/src/test/java/io/harness/cdng/artifact/resources/artifactory/service/ArtifactoryResourceServiceImplTest.java:349]: lets add unit tests for all- null, empty and inputPattern value.
lets add tests for both repository and imagePath so that we have coverage for all scenarios

**Review by Abhishek-CDC — APPROVED** [review]

**Review by puthrayaharness — APPROVED** [review]

**Comment by vivekkarnatiharness** [comment]
trigger uts

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Comment by vivekkarnatiharness** [comment]
trigger review

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck

**Comment by vivekkarnatiharness** [comment]
trigger messagecheck
